File: 0682/Sesión 08 - arquitectura/SoftInvBusiness/src/main/java/pe/edu/pucp/softinv/bo/AlmacenBO.java
package pe.edu.pucp.softinv.bo;

import pe.edu.pucp.softinv.dao.AlmacenDAO;
import pe.edu.pucp.softinv.daoImpl.AlmacenDAOImpl;
import pe.edu.pucp.softinv.model.AlmacenesDTO;

public class AlmacenBO {
    
    private AlmacenDAO almacenDAO;
    
    public AlmacenBO(){
        this.almacenDAO = new AlmacenDAOImpl();
    }
    
    public Integer insertar(String nombre, Boolean almacen_central){
        AlmacenesDTO almacen = new AlmacenesDTO();
        almacen.setNombre(nombre);
        almacen.setAlmacen_central(almacen_central);
        return this.almacenDAO.insertar(almacen);
    }
    
}

File: 0683/Sesión 11 - backend en C#/SoftInv/SoftInvBusiness/AlmacenBO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftInv.DAO;
using SoftInv.DAOImpl;
using SoftInv.Model;

namespace SoftInvBusiness
{
    public class AlmacenBO
    {
        private AlmacenDAO almacenDAO;

        public AlmacenBO()
        {
            this.almacenDAO = new AlmacenDAOImpl();
        }

        public int Insertar(string nombre, bool almacenCentral)
        {
            AlmacenesDTO almacenesDTO = new AlmacenesDTO();
            almacenesDTO.Nombre = nombre;
            almacenesDTO.AlmacenCentral = almacenCentral;
            return this.almacenDAO.Insertar(almacenesDTO);
        }
    }
}
